### PR TITLE
Estute/test android builds with make

### PR DESCRIPTION
**Pull request by estute**

### CODE REVIEW

**Comment by estute** [comment]
@jzoldak @benpatterson 
@BenjiLee just to keep you in the loop

File: mobileApp/jobs/buildWhiteLabelAndroidApps.groovy [hunk 1]
```diff
@@ -166,55 +169,90 @@ secretMap.each { jobConfigs ->
...
+                    reportFiles('lint-results-prodDebug.html')
+                    reportName('Linting Results')
+                }
+            }
 
+            // Publish to HockeyApp if user says so
             configure { project ->
-                project / publishers << 'org.jenkins__ci.plugins.flexible__publish.FlexiblePublisher' (plugin: 'flexible-publish@0.15.2') {
+                project / publishers << 'org.jenkins__ci.plugins.flexible__publish.FlexiblePublisher' (
+                    plugin: 'flexible-publish@0.15.2') {
```
Comment (benpatterson) [inline mobileApp/jobs/buildWhiteLabelAndroidApps.groovy]: Should this be captured somewhere more prominent?

File: mobileApp/jobs/buildWhiteLabelAndroidApps.groovy [hunk 2]
```diff
@@ -166,55 +169,90 @@ secretMap.each { jobConfigs ->
...
+                                    token "\${SHOULD_PUBLISH}"
+                                }
+                            // Manually configured XML for using the HockeyApp API, because it is not present in
+                            // the Jenkins Job DSL
                             publisherList {
                                 'hockeyapp.HockeyappRecorder' (schemaVersion: '2') {
                                     applications {
-                                        'hockeyapp.HockeyappApplication' (plugin: 'hockeyapp@1.2.1', schemaVersion: '1') {
+                                        'hockeyapp.HockeyappApplication' (
+                                            plugin: 'hockeyapp@1.2.1', schemaVersion: '1') {
```
Comment (benpatterson) [inline mobileApp/jobs/buildWhiteLabelAndroidApps.groovy]: Same here. If we change to a newer version of the plugin, will that mean we have to reprovision jobs?

File: mobileApp/resources/run_tests.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# Run emulator tests
+make e2e
+make artifacts
+
+# Again, kill all running emulators, as the Makefile spawns them as a background
+# process, and leaving them running can interfere with other jobs
+kill_all_emus
+
+
+exit 0
```
Comment (benpatterson) [inline mobileApp/resources/run_tests.sh]: If there is an error, will `set -e` result in an exit code even with this line here?

Also...is this line redundant anyway?

File: platform/jobs/edxPlatformUnitCoverage.groovy [hunk 4]
```diff
@@ -182,7 +182,7 @@ secretMap.each { jobConfigs ->
             }
             // Run jenkins-report.sh which will upload coverage results to
             // codecov.
-            shell("./scripts/jenkins-report.sh")
+            shell('./scripts/jenkins-report.sh')
```
Comment (benpatterson) [inline platform/jobs/edxPlatformUnitCoverage.groovy]: Ah groovy and the weird gString concept

File: testeng/jobs/backupJenkins.groovy [hunk 5]
```diff
@@ -125,7 +125,9 @@ secretMap.each { jobConfigs ->
         readFileFromWorkspace('testeng/resources/requirements.txt').split('\n').each { line ->
             script += "pip install --exists-action w ${line}\n"
         }
-        script += "aws ec2 create-snapshot --volume-id ${jobConfig['volumeId']} --description 'Automatic ${jobConfig['jenkinsInstance']} jenkins snapshot' > \${WORKSPACE}/snapshot-out.log\n"
+        script += "aws ec2 create-snapshot --volume-id ${jobConfig['volumeId']} " +
```
Comment (benpatterson) [inline testeng/jobs/backupJenkins.groovy]: Did you mean to put this change into this pr?

Comment (estute) [inline testeng/jobs/backupJenkins.groovy]: Yes, I was fixing codenarc violations to get it to pass

**Comment by estute** [comment]
@benpatterson I made a couple changes here:
1) I agree about the rigidity of the manually configured plugin. It will lead to trouble if we upgrade the plugin. Luckily, I saw there was DSL for part of the problem (flexible build plugin). I still need to address the HockeyAppPlugin...
2) I changed the run_tests.sh script because the Android SDK can sometimes make scripting tricky by appending a bunch of hidden characters into cmdline output.

**Comment by benpatterson** [comment]
👍

**Pull request merged**